Refactor/devops 384 harmonise ser

rmharrison opened the pull request:
If this PR fixes a bug, you _must_ add test cases representative of the bug.

#### What's this PR do?
See CHANGELOG.md

Briefly
- fixes linting
- Brings messaging up to date with auth/notifications (except removing gulp, which is only in auth)

#### Related JIRA tickets:
[DEVOPS-384](https://jira.amida.com/browse/DEVOPS-384)

#### How should this be manually tested?
`yarn jenkins` + POSTMAN

#### Any background context you want to provide?

#### Screenshots (if appropriate):

rmharrison committed:
[DEVOPS-384] Harmonise `package.json` and `yarn test`/`yarn jenkins`

rmharrison committed:
[DEVOPS-384] Resolve `yarn lint` errors

rmharrison committed:
[DEVOPS-384] Update `docker-compose.yml`

rmharrison committed:
[DEVOPS-384] Update AUTH_SERVICE defaults from v1 --> v2

rmharrison commented:
@nickmccally 
Test fails with same 5432 connection refused error [1](https://jenkins.amida.com/job/Messaging%20Microservice%20Unit%20Test/56/console) as https://github.com/amida-tech/amida-auth-microservice/pull/84

rmharrison commented:
Jenkins issue resolved by @nickmccally. `label:hold` removed.

unprofessional reviewed (approved):
- Code looks good and the tests pass via `yarn test`
- `docker-compose -f docker-compose.yml up` spins up the container + postgres with no problem
- Simple Postman GET request to `http://localhost:4001/api/v2/message/list?archived=false&limit=4&received=true` (with valid JWT Bearer token) returns expected results

unprofessional commented:
Actually, please see comment for https://github.com/amida-tech/amida-survey-microservice/pull/69#discussion_r263106088 regarding `.env.test` and/or `.env.production`

rmharrison committed:
Update `.env.test:MESSAGING_SERVICE_PG_DB`
`amida_messaging_microservice` --> `amida_messaging_microservice_tests`
Previously, `MESSAGING_SERVICE_PG_DB` was identical in both `.env.example` and `.env.test`

rmharrison committed:
npm:istanbul --> nyc

rmharrison commented:
Resolved out-of-date with rebased to tip of `branch:develop`
